esmtools/prediction.py

```diff
@@ -638,21 +707,24 @@ def PM_compute(ds, control, metric=_pearson_r, comparison=_m2m, anomaly=False,
         skill score
     """
     supervector_dim = 'svd'
+    comparison = _get_comparison_function(comparison)
     if comparison.__name__ not in ['_m2m', '_m2c', '_m2e', '_e2c']:
         raise ValueError('specify comparison argument')
 
+    metric = _get_metric_function(metric)
     if metric.__name__ in ['_pearson_r', '_rmse']:
         fct, truth = comparison(ds, supervector_dim)
         res = metric(fct, truth, dim=supervector_dim)
         return res
-    elif metric.__name__ in ['_mse', '_rmse_v', '_nrmse', '_nev', '_ppp', '_PPP', '_MSSS', '_uACC']:
+    elif metric.__name__ in ['_mse', '_rmse_v', '_nrmse', '_nev', '_ppp', '_PPP', 
+                             '_msss', '_uACC']:
         res = metric(ds, control, comparison, running)
         return res
     else:
         raise ValueError('specify metric argument')
 
 
-def DP_compute_skill(dp, reference, nlags=None):
+def compute_reconstruction(ds, reference, metric='pearson_r', nlags=None):
```

Comment:
Why call it reconstruction? 
We could add here the the comparison later. I’m not sure you will want to analyse also the skill of single starting years (still unsure how to do that) or the skill of ensemble members instead of just the ensemble mean forecast. 

If people want to have an ensemble mean forecast the could also just enter the mean into the function. Or there should be at least a flag to computer single member forecast and mean over the member dim.

Reply:
Maybe compute reference? Or just compute skill? Or just compute?

Reply:
> Why call it reconstruction?

I'll switch to compute_reference, but in the future we should have a PR that carefully looks at docstring language.

To me, CESM-DPLE is a hindcast-based prediction system. A hindcast being a model simulation of the past. However, this language is troublesome because the atmosphere community uses hindcast to mean actual predictions of the past.

So the language is better as "reconstruction" for a historical simulation. The CESM-DPLE reconstruction is forced by CORE (heat, momentum, freshwater flux) but has NO data assimilation. So this is not an assimilation either.

Make sense? I agree we go with "reference" because users can put in observations here to compute true skill against observations.